Source/ACE/Command/Handlers/DebugCommands.cs

```diff
@@ -825,8 +826,10 @@ public static void HandleWeapons(Session session, params string[] parameters)
             {
                 foreach (uint weenieId in weaponsTest)
                 {
-                    WorldObject loot = LootGenerationFactory.CreateTestWorldObject(session.Player, weenieId);
+                    AceObject lootAceObject = (AceObject)DatabaseManager.World.GetAceObjectByWeenie(weenieId).Clone(GuidManager.NewItemGuid().Full);
+                    WorldObject loot = new GenericObject(lootAceObject);
```

Comment:
I think the solution here is to make CreateNewWorldObject work.  CreateNewWorldObject is meant to be the primary tool for generating loot.  It currently only takes a weenie, but should probably be modified to accept a Loot Tier (int) as well.

This functionality should be an easy one liner, not a complicated daisy-chain of methods like Og had to implement.